lib/manageiq/messaging/client.rb

```diff
@@ -1,13 +1,42 @@
 module ManageIQ
   module Messaging
+    # The abstract client class. It defines methods needed to publish or subscribe messages.
+    # It is not recommended to directly create a solid subclass instance. The proper way is
+    # to call class method +Client.open+ with desired protocol. For example:
+    #
+    #   client = ManageIQ::Messaging::Client.open(
+    #     :protocol  => 'Stomp',
+    #     :host       => 'localhost',
+    #     :port       => 61616,
+    #     :password   => 'smartvm',
+    #     :username   => 'admin',
+    #     :client_ref => 'generic_1',
+    #     :encoding   => 'json'
+    #   )
+    #
+    # To close the connection one needs to explicitly call +client.close+.
+    # Alternately if a block is given for the +open+ method, the connection will be closed
```

Comment:
should be `Alternatively`, not `Alternately`.  See https://www.grammar.com/alternately_vs._alternatively